src/main/scala/inox/solvers/z3/Z3Native.scala

```diff
@@ -529,17 +543,19 @@ trait Z3Native extends ADTManagers with Interruptible { self: AbstractSolver =>
           if(ts.length > 4 && ts.substring(0, 2) == "bv" && ts.substring(ts.length - 4) == "[32]") {
             val integer = ts.substring(2, ts.length - 4)
             tpe match {
-              case Int32Type => IntLiteral(integer.toLong.toInt)
+              case Int8Type  => Int8Literal(integer.toLong.toByte) // FIXME is this right? above reads "ends with [32]"...
```

Comment:
Not sure about this piece of code: `[32]` seems to refer to 32-bit so maybe it just doesn't make sense to have other integer types handled here?